release/src/main/scripts/run_rc_validation.sh

```diff
@@ -100,6 +100,8 @@ HUB_ARTIFACTS_NAME=hub-linux-amd64-${HUB_VERSION}
ice" "beam-runners-jet" "beam-runners-local-java-core" "beam-runners-portability-java" "beam-runners-samza-job-server" "beam-runners-samza" "beam-runners-spark-3-job-server" "beam-runners-spark-3" "beam-runners-twister2" "beam-sdks-java-bom" "beam-sdks-java-core" "beam-sdks-java-expansion-service-app" "beam-sdks-java-expansion-service" "beam-sdks-java-extensions-arrow" "beam-sdks-java-extensions-avro" "beam-sdks-java-extensions-euphoria" "beam-sdks-java-extensions-google-cloud-platform-core" "beam-sdks-java-extensions-join-library" "beam-sdks-java-extensions-json-jackson" "beam-sdks-java-extensions-kryo" "beam-sdks-java-extensions-ml" "beam-sdks-java-extensions-protobuf" "beam-sdks-java-extensions-python" "beam-sdks-java-extensions-sbe" "beam-sdks-java-extensions-schemaio-expansion-service" "beam-sdks-java-extensions-sketching" "beam-sdks-java-extensions-sorter" "beam-sdks-java-extensions-sql-datacatalog" "beam-sdks-java-extensions-sql-expansion-service" "beam-sdks-java-extensions-sql-hcatalog" "beam-sdks-java-extensions-sql-jdbc" "beam-sdks-java-extensions-sql-udf" "beam-sdks-java-extensions-sql-zetasql" "beam-sdks-java-extensions-sql" "beam-sdks-java-extensions-timeseries" "beam-sdks-java-extensions-zetasketch" "beam-sdks-java-fn-execution" "beam-sdks-java-google-cloud-platform-bom" "beam-sdks-java-harness" "beam-sdks-java-io-amazon-web-services" "beam-sdks-java-io-amazon-web-services2" "beam-sdks-java-io-amqp" "beam-sdks-java-io-azure" "beam-sdks-java-io-cassandra" "beam-sdks-java-io-cdap" "beam-sdks-java-io-clickhouse" "beam-sdks-java-io-common" "beam-sdks-java-io-contextualtextio" "beam-sdks-java-io-csv" "beam-sdks-java-io-debezium-expansion-service" "beam-sdks-java-io-debezium" "beam-sdks-java-io-elasticsearch" "beam-sdks-java-io-expansion-service" "beam-sdks-java-io-file-schema-transform" "beam-sdks-java-io-google-cloud-platform-expansion-service" "beam-sdks-java-io-google-cloud-platform" "beam-sdks-java-io-hadoop-common" "beam-sdks-java-io-hadoop-file-system" "beam-sdks-java-io-hadoop-format" "beam-sdks-java-io-hbase" "beam-sdks-java-io-hcatalog" "beam-sdks-java-io-influxdb" "beam-sdks-java-io-jdbc" "beam-sdks-java-io-jms" "beam-sdks-java-io-kafka" "beam-sdks-java-io-kinesis-expansion-service" "beam-sdks-java-io-kinesis" "beam-sdks-java-io-kudu" "beam-sdks-java-io-mongodb" "beam-sdks-java-io-mqtt" "beam-sdks-java-io-neo4j" "beam-sdks-java-io-parquet" "beam-sdks-java-io-pulsar" "beam-sdks-java-io-rabbitmq" "beam-sdks-java-io-redis" "beam-sdks-java-io-singlestore" "beam-sdks-java-io-snowflake-expansion-service" "beam-sdks-java-io-snowflake" "beam-sdks-java-io-solr" "beam-sdks-java-io-sparkreceiver-2" "beam-sdks-java-io-splunk" "beam-sdks-java-io-synthetic" "beam-sdks-java-io-thrift" "beam-sdks-java-io-tika" "beam-sdks-java-io-xml" "beam-sdks-java-maven-archetypes-examples" "beam-sdks-java-maven-archetypes-gcp-bom-examples" "beam-sdks-java-maven-archetypes-starter" "beam-sdks-java-nexmark" "beam-sdks-java-test-utils" "beam-sdks-java-tpcds" )
```

Comment:
Possibly we can auto-generate this list instead of hard coding a manually generated list ? Otherwise this will end up become stale very quickly and trying to manually verify/update this list can be very time consuming.

Reply:
If you have ideas on autogeneration, I would definitely prefer that route

Reply:
Can we locally push the set of jars in to a clean repo using "./gradlew publish" (same command we use during releases) and update this list ?

We could create a script for this and create a Gradle build target (but probably this is slow so we should not generate in the default build target).

Reply:
I think relying on the same command we're using to create the release is an antipattern; if something breaks with our publish infra (e.g. a gradle target's publish somehow is broken), that would no longer be caught

Reply:
Ah, I though the purpose of this check was to make sure that all jars were uploaded to the Sonatype repo correctly not to check whether the "./gradlew publish" is broken somehow. I think latter is very unlikely BTW and I think the likelihood of this manual list being out of sync will be higher than the Gradle command being broken.

Reply:
Historically "missing jars in Sonatype repo" kind of issues were due to network issues (for e.g., IP address changing during upload) not due to issues with the build itself.

Reply:
That said, if you want to get the validation in the current form I'm fine with it :)

Reply:
My goal was to test both (the goal is to test whether all of the jars that we expect to get uploaded did indeed get uploaded, independent of implementation). With that said, you're probably right that this isn't a priority to test anything other than the network upload.

I updated to use the java-bom to get the artifacts we need to upload. This should have all of the jars we upload as dependencies, and I validated it correctly checks everything